File: src/StreamAPII/Person.java
package StreamAPII;

public class Person {
    private String name;
    private int age;

    Person(String name, int age){
        this.age = age;
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }
}

File: Module4/4.8.AbstractionClasses/src/Person.kt
abstract class Person(val name : String, val lastName: String, age : Int) {
    var age = age
        get() = field
        private set

    fun birthDay(){
        age++
        println("У человека день рождения. Теперь ему $age")
    }

    abstract fun action()
    abstract fun firing()
}

class Student(name: String, lastName: String, age: Int) : Person(name, lastName, age) {

    override fun action() {
        println("Студент $name $lastName сидит на паре")
    }

    override fun firing() {
        println("Студент $name $lastName отчислен")
    }

}

class Employee(name: String, lastName: String, age: Int) : Person(name, lastName, age) {
    override fun action() {
        println("Сотрудник $name $lastName работает")
    }

    override fun firing() {
        println("Сотрудник $name $lastName уволен")
    }
}